pkg/apis/eventing/v1alpha1/channel_types.go

```diff
@@ -0,0 +1,172 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	"encoding/json"
+
+	"github.com/knative/pkg/apis"
+	"github.com/knative/pkg/webhook"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Channel represents a named endpoint on which a Bus accepts event delivery and
+// corresponds to the channels.channels.knative.dev CRD. The Bus handles
+// provisioning channels, delivering events to Channels, and delivering events
+// from Channels to their Subscriptions.
+type Channel struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the Channel.
+	Spec ChannelSpec `json:"spec,omitempty"`
+
+	// Status represents the current state of the Channel. This data may be out of
+	// date.
+	// +optional
+	Status ChannelStatus `json:"status,omitempty"`
+}
+
+// Check that Channel can be validated, can be defaulted, and has immutable fields.
+var _ apis.Validatable = (*Channel)(nil)
+var _ apis.Defaultable = (*Channel)(nil)
+var _ apis.Immutable = (*Channel)(nil)
+var _ runtime.Object = (*Channel)(nil)
+var _ webhook.GenericCRD = (*Channel)(nil)
+
+// ChannelSpec specifies the Provisioner backing a channel and the configuration
+// arguments for a Channel.
+type ChannelSpec struct {
+	// TODO: Generation does not work correctly with CRD. They are scrubbed
+	// by the APIserver (https://github.com/kubernetes/kubernetes/issues/58778)
+	// So, we add Generation here. Once that gets fixed, remove this and use
+	// ObjectMeta.Generation instead.
+	// +optional
+	Generation int64 `json:"generation,omitempty"`
+
+	// Provisioner defines the name of the Provisioner backing this channel.
+	// TODO: +optional If missing, a default Provisioner may be selected for the Channel.
+	Provisioner *ProvisionerReference `json:"provisioner,omitempty"`
+
+	// Arguments defines the arguments to pass to the Provisioner which provisions
+	// this Channel.
+	// +optional
+	Arguments *runtime.RawExtension `json:"arguments,omitempty"`
+}
+
+// ChannelStatus represents the current state of a Channel.
+type ChannelStatus struct {
+	// ObservedGeneration is the most recent generation observed for this Channel.
+	// It corresponds to the Channel's generation, which is updated on mutation by
+	// the API Server.
+	//TODO The above comment is only true once
+	// https://github.com/kubernetes/kubernetes/issues/58778 is fixed.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// A reference to the k8s Service backing this channel.
+	// +optional
+	Service *corev1.LocalObjectReference `json:"service,omitempty"`
```

Comment:
There was a notion of capturing all provisioned resources in a generic way. Perhaps `Provisioned []corev1.ObjectReference`. Likely, this is overkill if those resources have OwnerReferences back to this Channel.

Service and VirtualService are too implementation specific and should be removed

Reply:
Removed these fields. We can add `Provisioned` later if owner references prove insufficient.